src/transformers/retrieval_rag.py

```diff
@@ -236,15 +239,19 @@ def init_index(self):
                 index_name=self.index_name,
                 dummy=self.use_dummy_dataset,
             )
+            self.dataset.set_format("numpy", columns=["embeddings"], output_all_columns=True)
         self._index_initialize = True
 
     def get_doc_dicts(self, doc_ids: np.ndarray) -> List[dict]:
         return [self.dataset[doc_ids[i].tolist()] for i in range(doc_ids.shape[0])]
 
     def get_top_docs(self, question_hidden_states: np.ndarray, n_docs=5) -> Tuple[np.ndarray, np.ndarray]:
-        _, docs = self.dataset.get_nearest_examples_batch("embeddings", question_hidden_states, n_docs)
-        ids = [[int(i) for i in doc["id"]] for doc in docs]
+        _, ids = self.dataset.search_batch("embeddings", question_hidden_states, n_docs)
+        docs = [self.dataset[[i for i in indices if i >= 0]] for indices in ids]
         vectors = [doc["embeddings"] for doc in docs]
+        for i in range(len(vectors)):
+            if len(vectors[i]) < n_docs:
+                vectors[i] = np.vstack([vectors[i], np.zeros((n_docs - len(vectors[i]), self.vector_size))])
```

Comment:
I added these lines because there exist some indexes that could return ids with -1.
In that case I just pad the vectors with zeros.